pkgs/top-level/perl-packages.nix

```diff
@@ -10968,6 +10969,11 @@ let
         shortenPerlShebang "$file"
       done
     '';
+    postInstall = ''
+      for file in latexmlc latexmlmath latexmlpost ; do
+        wrapProgram $out/bin/$file --prefix PATH : ${lib.makeBinPath [ pkgs.ghostscript pkgs.potrace ]}
```

Comment:
We build ImageMagick with `--with-frozenpaths`, so should it not already hardcode the full paths in delegates xml?

Reply:
That's a fair question. The `imagemagick` package does not include every runtime dependency among the `buildInputs`, not even `imagemagickBig`. On top of that, `PerlMagick` only depends on `imagemagick`, not `imagemagickBig`.

I suppose we could override PerlMagick and imagemagick to add the two dependencies locally, would that be more reasonable? Possibly a bit wasteful in terms of space.

Reply:
After some thinking, I suggest we override PerlMagick to use imagemagickBig, so that imagemagick is compiled with gslib (ikiwiki is already doing this, for instance); in theory, PerlMagick won't call `gs` anymore and will be a bit faster for postscript operations.

But potrace is a genuine runtime dependency, not a build dependency, my feeling is that it should be treated as such using `wrapProgram`.

If there is a perl specific version of `wrapProgram`, I'd be happy to use that. It seems like the `--argv0` option has no effect when used on perl scripts. It doesn't break anything, but it is ugly to see `.latexmlpost-wrapped` instead of `latexmlpost`.

Reply:
@veprbl ...is this too roundabout?
```nix
propagatedBuildInputs = [ ArchiveZip DBFile FileWhich IOString ImageSize JSONXS LWP ParseRecDescent PodParser TextUnidecode XMLLibXSLT
  (pkgs.perlPackages.override { pkgs = pkgs // { imagemagick = pkgs.imagemagickBig; }; }).PerlMagick ];
```
I confirm it does the job for the ghostscript dependency, but this would make LaTeXML the first and only package overriding all of perlPackages *within* perlPackages. And yet I can't see another self contained way – you can't simply override `buildInputs` in PerlMagick, you would have to rewrite most of its derivation.

What about a new PerlMagickBig (separate PR of course)? Even just to ensure that ikiwiki and LaTeXML end up using the same PerlMagick. I can't quite tell if they are getting different copies of the overridden PerlMagick.

It appears I am once again out of my depth. I really thought this would be a simple patch...

Reply:
I don't think that letting deep overrides into perl-packages.nix is a good idea. At this point I'm somewhat tempted to have this moved out from perl-packages, assuming that this is mostly an application, not a library. I think we can merge this PR as is for now (perhaps might want to toss `texlive.bin.dvipng` in to the PATH as well).

cc @stigtsp

Reply:
> assuming that this is mostly an application, not a library.

Yeah, the [documentation](https://dlmf.nist.gov/LaTeXML/manual/usage/) doesn't really suggest you should be using it as a library. The few projects I am aware of use the binaries, I think, not the modules.

> (perhaps might want to toss `texlive.bin.dvipng` in to the PATH as well).

That's a very good idea, I'll do some tests and report back or update the PR. And I'll try again with `texlive.bin.dvisvgm`, I was too busy figuring out ImageMagick to make that work.

Reply:
> > assuming that this is mostly an application, not a library.
> 
> Yeah, the [documentation](https://dlmf.nist.gov/LaTeXML/manual/usage/) doesn't really suggest you should be using it as a library. The few projects I am aware of use the binaries, I think, not the modules.

Formally, there is the documentation for the module https://metacpan.org/pod/LaTeXML

Reply:
I think moving it out of `perl-packages.nix` is a good idea if an override, or other non-trivial change is needed. Otherwise it looks OK to me.

Reply:
By the way, my test was quick and unsuccessful – `dvipng` fails just like `dvisvgm`. They both require pieces of TeX Live (fonts!) which I don't think should be hard coded in LaTeXML. Better to let LaTeXML work with the user's TeX Live installation, and leave the PR at the very minimum.

Reply:
Given the exchange about `dvipng` and `dvisvgm`, I have added a short comment defining which dependencies should be added via `wrapProgram`. I can squash everything in a single commit.

Reply:
Yes, pls squash it.

Reply:
> I'm somewhat tempted to have this moved out from perl-packages

@veprbl before I open a useless issue/PR, in case I misunderstand `nix-env -u`. Would this change how LaTeXML gets updated?

I gather that `nix-env -u` stops updating LaTeXML when the underlying perl version disappears from nixpkgs. Have I got this right? If that's the case, a derivation with pname `LaTeXML` would make more sense, especially because LaTeXML has a slow release cycle, about once a year. I think I hit this problem last year with perl5.28, I had to uninstall and reinstall to get the update (not entirely sure though – that was too long ago).

I can open an issue for discussion, or directly a PR, _if_ my understanding of versioning is correct. (Also `nix-env -i latexml` would be more user-friendly... but that can be done independently of this.)

Reply:
@xworld21 
Yes, I believe you are correct, the updates via `nix-env -u` will not change the perl version as it is part of the package name:
```
# nix eval --impure --expr '(builtins.parseDrvName (import <nixpkgs> {}).perlPackages.LaTeXML.name).name'
"perl5.32.1-LaTeXML"
```
It seems like `nix profile` operates with attribute names.

Reply:
@veprbl thanks! So a `latexml` package seems sensible. I'll run some experiments to see if I can get `latexml-0.8.5` but also keep the current perl-packages.nix arrangement (there are users already after all), and come back with an issue or a tentative PR.